src/interface_adapter/login/LoginController.java

```diff
@@ -2,18 +2,26 @@
 
 import use_case.login.LoginInputBoundary;
 import use_case.login.LoginInputData;
+import use_case.switch_view.SwitchViewInputBoundary;
+import use_case.switch_view.SwitchViewInputData;
 
 public class LoginController {
 
   final LoginInputBoundary loginUseCaseInteractor;
+  private final SwitchViewInputBoundary switchViewInteractor;
 
-  public LoginController(LoginInputBoundary loginUseCaseInteractor) {
+  public LoginController(LoginInputBoundary loginUseCaseInteractor, SwitchViewInputBoundary switchViewInteractor) {
     this.loginUseCaseInteractor = loginUseCaseInteractor;
+    this.switchViewInteractor = switchViewInteractor;
   }
 
   public void execute(String username, String password) {
     LoginInputData loginInputData = new LoginInputData(username, password);
 
     loginUseCaseInteractor.execute(loginInputData);
   }
+
+  public void switchToSignup() {
+    switchViewInteractor.execute(new SwitchViewInputData("sign up"));
```

Comment:
Magic string, see my question on using statics as a different way to do this https://github.com/uoftm/csc207project/pull/6

Reply:
Let's meet as a team to discuss this. I think it's ok for now